tools/load_balancing_tool/load_balancing_submit.py

```diff
@@ -228,97 +222,94 @@ def load_balancing_submit(compset, res, pesfile, compiler, project, machine,
             logger.critical('No pesize for pes node in file %s', pesfile)
         if pesize in pesize_list:
             logger.critical('pesize %s duplicated in file %s', pesize, pesfile)
-        logger.info('  ' + pesize)
         pesize_list.append(pesize)
 
     if not pesize_list:
         logger.critical('ERROR: No grid entries found in pes file %s', pesfile)
         raise SystemExit(1)
 
-    # Submit job for each entry in pesfile
-    script_dir = CIME.utils.get_scripts_root()
-    for pesize in pesize_list:
-        casename = casename_prefix + pesize
-        casedir = os.path.join(script_dir, casename)
-        logger.info('Case %s...', casename)
-        if os.path.exists(casedir):
+    machobj = Machines(machine=machine)
+    if test_root is None:
+        test_root = machobj.get_value("CIME_OUTPUT_ROOT")
+    if machine is None:
+        machine = machobj.get_machine_name()
+        print "machine is {}".format(machine)
+    if compiler is None:
+        compiler = machobj.get_default_compiler()
+        print "compiler is {}".format(compiler)
+    if mpilib is None:
+        mpilib = machobj.get_default_MPIlib()
+
+
+    test_names = []
+    for i in xrange(len(pesize_list)):
+        test_names.append("PFS_I{}.{}.{}.{}_{}".format(i, res, compset, machine, compiler))
```

Comment:
This should use the standard API for creating a "full" test name: update_acme_tests.get_full_test_names()